### PR TITLE
Cdsb

### PR DESCRIPTION
**Pull request by Adam-D-Lewis**
Adds support for other visualization providers supported by cdsdashboards,
resolves https://github.com/Quansight/qhub-onprem/issues/36

### CODE REVIEW

File: files/environments/jupyterlab.yaml [hunk 1]
```diff
@@ -13,9 +13,17 @@ dependencies:
   # dask
   - dask-gateway
   - dask-labextension>=5.0.0
-  # dashboard
+  # panel
+  - jupyter-panel-proxy
```
Comment (costrouc) [inline files/environments/jupyterlab.yaml:17]: Could you remove jupyter-panel-proxy? Tried to remove this in favor of your work.

File: files/environments/jupyterlab.yaml [hunk 2]
```diff
@@ -13,9 +13,17 @@ dependencies:
...
+  # required to launch jupterlab from jupyterhub
+  - jupyterhub
+  # dashboards (versions specified to narrow conda solve space)
   - cdsdashboards-singleuser
-  - panel
```
Comment (costrouc) [inline files/environments/jupyterlab.yaml:18]: Add panel as dependency

File: templates/jupyterhub_config.py.j2 [hunk 3]
```diff
@@ -34,22 +34,19 @@ c.JupyterHub.extra_handlers = []
...
-                '{--}allow-websocket-origin={origin_host}',
-                '--ready-check-path=/ready-check']
-        }
-}
-c.CDSDashboardsConfig.extra_presentation_types = ['qhub-bokeh']
+from cdsdashboards.hubextension.spawners.variablemixin import VariableMixin, MetaVariableMixin, Command
 
 class QHubHPCSpawner(QHubHPCSpawner, VariableMixin, metaclass=MetaVariableMixin):
-    pass
+    default_presentation_cmd = Command(
```
Comment (costrouc) [inline templates/jupyterhub_config.py.j2:40]: Not needed. Can replace 

```
default_presentation_cmd = Command(
    ['jhsingle-native-proxy'],
        allow_none=False,
        help="""
        The command to run presentations through jhsingle-native-proxy, can be a string or list.
        Default is ['jhsingle-native-proxy'].
        """
    ).tag(config=True)
```

with

```python
c.VariableMixin.default_presentation_cmd = ['{{ miniforge.home }}/envs/{{ jupyterhub.lab_environment }}/bin/jhsingle-native-proxy']
```

**Review by costrouc — CHANGES_REQUESTED** [review]

**Comment by costrouc** [comment]
Was able to test PR and it worked. Minor fixes and then ready for merge.

File: files/environments/jupyterlab.yaml (hunk 2 above)
Comment (Adam-D-Lewis) [reply files/environments/jupyterlab.yaml:18]: panel is a dependency below (I added a version to narrow conda solve space)

Comment (costrouc) [reply files/environments/jupyterlab.yaml:18]: Whoops sorry about that!

**Review by costrouc — APPROVED** [review]

**Comment by costrouc** [comment]
Looks good to me. Thanks!

**Pull request merged**